ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -250,6 +242,18 @@ class SeparateOutXAxisKernel : public asset::CFloatingPointSeparableImageFilterK
         );
     }
 
+    auto CMitsubaMaterialCompilerFrontend::getDerivMap(asset::ICPUImage* _heightMap, asset::ICPUSampler* _smplr) -> deriv_map_cache_val_t
+    {
+        auto key = deriv_map_cache_key_t{core::smart_refctd_ptr<asset::ICPUImage>(_heightMap), core::smart_refctd_ptr<asset::ICPUSampler>(_smplr)};
+        auto found = m_derivMapCache.find(key);
+        if (found != m_derivMapCache.end())
+            return found->second;
+
+        auto map = createDerivMap(_heightMap, _smplr);
+        m_derivMapCache.insert({ std::move(key), map });
```

Comment:
this concern the MTL loader too

You need to check `_heightMap` is not dummy, if it is restore it, etc. also cache the derivative map in the general cache under a custom key... and so on